semantic-python/src/Language/Python/Core.hs

```diff
@@ -203,7 +203,18 @@ instance Compile Py.Block where
 
 instance Compile Py.BooleanOperator
 instance Compile Py.BreakStatement
-instance Compile Py.Call
+
+instance Compile Py.Call where
+  compileCC it@Py.Call
+    { function
+    , arguments = L1 Py.ArgumentList { extraChildren = args }
+    } cc = do
+    func <- compileCC function cc
+    -- Python function arguments are defined to evaluate left to right.
+    args <- traverse (flip compileCC cc) args
```

Comment:
This (and line 212) is wrong:

1. We definitely shouldn’t transfer control to the arguments (passing `cc` to them).
2. We _definitely_ definitely shouldn’t transfer control more than once (passing `cc` to multiple things).

We want to use the continuation once and only once, after `func $$* args`.

Instead, we want to compile all of these subterms as expressions—i.e. give them a “pure” continuation that simply returns their values. I’m not sure if we finished that approach?

Reply:
(To illustrate the wrongness, try a test case where you do something after a function call with multiple arguments.)

Reply:
This should be fixed now with the power gained from #320.